extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java

```diff
@@ -207,13 +207,30 @@ public void build(ApplicationArchivesBuildItem archivesBuildItem,
             annotationModel = null;
         }
         OpenApiDocument finalDocument = loadDocument(staticModel, annotationModel);
+        String basePath = getOpenAPIBasePath() + OpenApiServlet.GENERATED_DOC_BASE;
         for (OpenApiSerializer.Format format : OpenApiSerializer.Format.values()) {
-            String name = OpenApiServlet.GENERATED_DOC_BASE + format;
+            String name = basePath + format;
             resourceBuildItemBuildProducer.produce(new GeneratedWebResourceBuildItem(name,
                     OpenApiSerializer.serialize(finalDocument.get(), format).getBytes(StandardCharsets.UTF_8)));
         }
     }
 
+    private String getOpenAPIBasePath() {
+        List<String> pathSegments = Arrays.asList(openapi.path.split(OPEN_API_PATH_SEPARATOR))
+                .parallelStream()
+                .filter(s -> !s.isEmpty())
+                .collect(Collectors.toList());
+
+        if (pathSegments.size() <= 1) {
```

Comment:
Doesn't this mean if no value is set, and thus the default config property is used, that we return "" instead of "/openapi" as the default only has one segment?

I also think a test case should be added for this situation, ie. no value set in `application.properties`

Reply:
> I also think a test case should be added for this situation, ie. no value set in `application.properties`

I have added the missing test case for the default config.